File: src/main/java/net/gegy1000/terrarium/server/map/adapter/RegionAdapter.java
package net.gegy1000.terrarium.server.map.adapter;

import net.gegy1000.terrarium.server.map.RegionData;
import net.gegy1000.terrarium.server.world.EarthGenerationSettings;

public interface RegionAdapter {
    void adapt(EarthGenerationSettings settings, RegionData data, int x, int z, int width, int height);
}

File: src/main/kotlin/net/gegy1000/terrarium/server/map/adapter/RegionAdapter.kt
package net.gegy1000.terrarium.server.map.adapter

import net.gegy1000.terrarium.server.map.glob.GlobType
import net.gegy1000.terrarium.server.map.source.osm.OverpassTileAccess
import net.gegy1000.terrarium.server.world.EarthGenerationSettings

interface RegionAdapter {
    fun adaptGlobcover(settings: EarthGenerationSettings, overpassTile: OverpassTileAccess, globBuffer: Array<GlobType>, x: Int, y: Int, width: Int, height: Int)
}
